network/wakeonlan.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2016, Dag Wieers <dag@wieers.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: wakeonlan
+version_added: 2.2
+short_description: Send magic Wake-on-LAN (WoL) broadcast packet
+description:
+   - The M(wakeonlan) module sends magic Wake-on-LAN (WoL) broadcast packets.
+options:
+  mac:
+    description:
+      - MAC address to send wake-on-lan broadcast packet for
+    required: true
+    default: null
+  broadcast:
+    description:
+      - Network broadcast address to use for broadcasting
+    required: false
+    default: 255.255.255.255
+  port:
+    description:
+      - UDP port to use for magic packet
+    required: false
+    default: 7
+author: "Dag Wieers (@dagwieers)"
+todo:
```

Comment:
Related to the arping support, this may not be possible in a pure-python fashion. Which would then boil down to either requiring arping command support, or conditional check-mode support. Which are both hard pills to swallow IMO :-/